src/biogeophys/QSatMod.F90

```diff
@@ -12,58 +12,56 @@ module QSatMod
-    real(r8), parameter :: c7 =  0.149436277e-11_r8
-    real(r8), parameter :: c8 =  0.262655803e-14_r8
-    ! For derivative:ice
-    real(r8), parameter :: d0 =  0.503277922_r8
-    real(r8), parameter :: d1 =  0.377289173e-01_r8
-    real(r8), parameter :: d2 =  0.126801703e-02_r8
-    real(r8), parameter :: d3 =  0.249468427e-04_r8
-    real(r8), parameter :: d4 =  0.313703411e-06_r8
-    real(r8), parameter :: d5 =  0.257180651e-08_r8
-    real(r8), parameter :: d6 =  0.133268878e-10_r8
-    real(r8), parameter :: d7 =  0.394116744e-13_r8
-    real(r8), parameter :: d8 =  0.498070196e-16_r8  
-contains
-
+  ! For water vapor (temperature range 0C-100C)
+  real(r8), parameter :: a0 =  6.11213476_r8
+  real(r8), parameter :: a1 =  0.444007856_r8
+  real(r8), parameter :: a2 =  0.143064234e-01_r8
+  real(r8), parameter :: a3 =  0.264461437e-03_r8
+  real(r8), parameter :: a4 =  0.305903558e-05_r8
+  real(r8), parameter :: a5 =  0.196237241e-07_r8
+  real(r8), parameter :: a6 =  0.892344772e-10_r8
+  real(r8), parameter :: a7 = -0.373208410e-12_r8
+  real(r8), parameter :: a8 =  0.209339997e-15_r8
+  ! For derivative:water vapor
+  real(r8), parameter :: b0 =  0.444017302_r8
+  real(r8), parameter :: b1 =  0.286064092e-01_r8
+  real(r8), parameter :: b2 =  0.794683137e-03_r8
+  real(r8), parameter :: b3 =  0.121211669e-04_r8
+  real(r8), parameter :: b4 =  0.103354611e-06_r8
+  real(r8), parameter :: b5 =  0.404125005e-09_r8
+  real(r8), parameter :: b6 = -0.788037859e-12_r8
+  real(r8), parameter :: b7 = -0.114596802e-13_r8
+  real(r8), parameter :: b8 =  0.381294516e-16_r8
+  ! For ice (temperature range -75C-0C)
+  real(r8), parameter :: c0 =  6.11123516_r8
+  real(r8), parameter :: c1 =  0.503109514_r8
+  real(r8), parameter :: c2 =  0.188369801e-01_r8
+  real(r8), parameter :: c3 =  0.420547422e-03_r8
+  real(r8), parameter :: c4 =  0.614396778e-05_r8
+  real(r8), parameter :: c5 =  0.602780717e-07_r8
+  real(r8), parameter :: c6 =  0.387940929e-09_r8
+  real(r8), parameter :: c7 =  0.149436277e-11_r8
+  real(r8), parameter :: c8 =  0.262655803e-14_r8
+  ! For derivative:ice
+  real(r8), parameter :: d0 =  0.503277922_r8
+  real(r8), parameter :: d1 =  0.377289173e-01_r8
+  real(r8), parameter :: d2 =  0.126801703e-02_r8
+  real(r8), parameter :: d3 =  0.249468427e-04_r8
+  real(r8), parameter :: d4 =  0.313703411e-06_r8
+  real(r8), parameter :: d5 =  0.257180651e-08_r8
+  real(r8), parameter :: d6 =  0.133268878e-10_r8
+  real(r8), parameter :: d7 =  0.394116744e-13_r8
+  real(r8), parameter :: d8 =  0.498070196e-16_r8
 
+contains
 
   !-----------------------------------------------------------------------
-  subroutine QSat (T, p, es, esdT, qs, qsdT)
+  subroutine QSat (T, p, qs, es, qsdT, esdT)
     !
     ! !DESCRIPTION:
-    ! Computes saturation mixing ratio and the change in saturation
+    ! Computes saturation mixing ratio and (optionally) the change in saturation
     ! mixing ratio with respect to temperature.
```

Comment:
@billsacks all your mods make sense to me. Thank you for introducing this simplification and enhanced flexibility to subroutine QSat.

I have a suggested addition to the above comment. I had forgotten that mixing ratio and specific humidity are treated as equal in meteorology, and I may not be alone in forgetting:
```suggestion
    ! mixing ratio with respect to temperature. Mixing ratio and 
    specific humidity are approximately equal and can be treated 
    as the same.
```

Reply:
Done - thanks